src/main/java/xyz/nucleoid/extras/util/GameVersionResolver.java

```diff
@@ -0,0 +1,40 @@
+package xyz.nucleoid.extras.util;
+
+import com.viaversion.viaversion.api.Via;
+import com.viaversion.viaversion.api.protocol.version.ProtocolVersion;
+
+import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.SharedConstants;
+
+public final class GameVersionResolver {
+    private static final boolean VIAVERSION = FabricLoader.getInstance().isModLoaded("viaversion");
+
+    private GameVersionResolver() {
+        return;
+    }
+
+    public static String getVersion() {
+        if (VIAVERSION) {
+            String version = getViaVersionHighestSupportedVersion();
+            if (version != null) {
+                return version;
+            }
+        }
+
+        return SharedConstants.getGameVersion().getName();
+    }
+
+    private static String getViaVersionHighestSupportedVersion() {
```

Comment:
Nit: `@Nullable`?

Reply:
Fixed in a7b00ab7fdcf8c43040fdfd370f88a7d3037c77f